core/src/main/java/tachyon/worker/netty/BlockResponse.java

```diff
@@ -69,26 +69,29 @@ protected void encode(final ChannelHandlerContext ctx, final BlockResponse msg,
       // Add the header information to output
       out.add(createHeader(ctx, msg));
 
-      BlockHandler handler = msg.getHandler();
-      if (handler != null) {
-        FileTransferType type = mTachyonConf.getEnum(Constants.WORKER_NETTY_FILE_TRANSFER_TYPE,
-            FileTransferType.TRANSFER);
-        switch (type) {
-          case MAPPED:
-            ByteBuffer data = handler.read(msg.getOffset(), (int) msg.getLength());
-            out.add(Unpooled.wrappedBuffer(data));
+      final BlockHandler handler = msg.getHandler();
+      if (handler == null) {
+        return;
+      }
+      final FileTransferType type =
+          mTachyonConf
+              .getEnum(Constants.WORKER_NETTY_FILE_TRANSFER_TYPE, FileTransferType.TRANSFER);
+      switch (type) {
+        case MAPPED:
+          ByteBuffer data = handler.read(msg.getOffset(), (int) msg.getLength());
+          out.add(Unpooled.wrappedBuffer(data));
+          handler.close();
+          break;
+        case TRANSFER:
+        default:
+          if (handler.getChannel() instanceof FileChannel) {
+            out.add(new DefaultFileRegion((FileChannel) handler.getChannel(), msg.getOffset(), msg
+                .getLength()));
```

Comment:
Does the formatter suggest any line break style? This feels kind of weird to me.